Code/SimDivPickers/testPickers.cpp

```diff
@@ -37,19 +37,27 @@ void testGithub2245() {
   BOOST_LOG(rdErrorLog) << "Testing github issue 2245: MinMax Diversity picker "
                            "seeding shows deterministic / non-random behaviour."
                         << std::endl;
+  const int MAX_ALLOWED_FAILURES = 3;
+  int maxAllowedFailures;
   {
     RDPickers::MaxMinPicker pkr;
     int poolSz = 1000;
     auto picks1 = pkr.lazyPick(dist_on_line, poolSz, 10, RDKit::INT_VECT(), -1);
-    auto picks2 = pkr.lazyPick(dist_on_line, poolSz, 10, RDKit::INT_VECT(), -1);
-    TEST_ASSERT(picks1 != picks2);
+	for (maxAllowedFailures = MAX_ALLOWED_FAILURES; maxAllowedFailures; --maxAllowedFailures) {
```

Comment:
Might look clearer without the loop?

    auto picks1 = pkr.lazyPick(dist_on_line, poolSz, 10, RDKit::INT_VECT(), -1);
    auto picks2 = pkr.lazyPick(dist_on_line, poolSz, 10, RDKit::INT_VECT(), -1);
    auto picks3 = pkr.lazyPick(dist_on_line, poolSz, 10, RDKit::INT_VECT(), -1);
    TEST_ASSERT(picks1 != picks2 || picks1 != picks3);

Reply:
@d-b-w : you'd need to add `picks4` too. At some point the loop is clearer.